qt/applications/workbench/workbench/app/mainwindow.py

```diff
@@ -414,10 +424,54 @@ def open_file(self):
             return
         self.editor.open_file_in_new_tab(filepath)
 
-    def save_file(self):
-        # todo: how should this interact with project saving and workspaces when they are implemented?
+    def save_script(self):
         self.editor.save_current_file()
 
+    def save_project(self):
+        if self.last_project_location is None:
+            self.save_project_as()
+        else:
+            # Clear directory before saving to remove old workspaces
+            files = glob.glob(self.last_project_location + '/.*')
+            for f in files:
+                os.remove(f)
```

Comment:
We need to be careful when removing files as it's entirely possible the OS can't delete the file for some reason and will raise an `OSError`. We should not fall over in this case.

Reply:
Handled properly! :+1: